packages/files/src/client.ts

```diff
@@ -0,0 +1,243 @@
+        dv: 1
+        fingerprint: string
+    }
+    id: string
+    user: {
+        id: string
+    }
+    fileClientId: string
+    modelNames?: string[]
+}
+
+export interface ShareOptions {
+    serverUrl?: string
+    cookie?: string
+    payload: SharePayload
+}
+
+export interface AttachPayload {
+    dv: 1
+    sender: {
+        dv: 1
+        fingerprint: string
+    }
+    modelName: string
+    itemId: string
+    fileClientId: string
+    signature: string
+}
+
+export interface AttachOptions {
+    serverUrl?: string
+    cookie?: string
+    payload: AttachPayload
+}
+
+export interface UploadFileResult {
+    id: string
+    signature: string
+    attached?: boolean
+    publicSignature?: string
+}
+
+export interface UploadResponse {
+    files: UploadFileResult[]
+}
+
+export interface ShareResponse {
+    file: {
+        id: string
+        signature: string
+    }
+}
+
+export interface AttachResponse {
+    file: {
+        signature: string
+    }
+}
+
+export function buildMeta ({
+    userId,
+    fileClientId,
+    modelNames,
+    fingerprint,
+    organizationId,
+}: BuildMetaOptions): FileMeta {
+    const meta: FileMeta = {
+        dv: 1,
+        sender: { dv: 1, fingerprint },
+        user: { id: userId },
+        fileClientId,
+        modelNames,
+    }
+    if (organizationId) {
+        meta.organization = { id: organizationId }
+    }
+    return meta
+}
+
+/**
+ * Get server URL from environment variables or use provided override.
+ */
+function getServerUrl (override?: string): string {
+    if (override) {
+        return override.replace(/\/$/, '')
+    }
+
+    if (typeof window !== 'undefined') {
+        const envUrl = (typeof process !== 'undefined' && process.env?.NEXT_PUBLIC_SERVER_URL) ||
+                       (window as any).__NEXT_DATA__?.env?.NEXT_PUBLIC_SERVER_URL
+        if (envUrl) {
+            return envUrl.replace(/\/$/, '')
+        }
+        return window.location.origin
+    } else {
+        const envUrl = process.env.SERVER_URL || process.env.FILE_SERVER_URL
+        if (envUrl) {
+            return envUrl.replace(/\/$/, '')
+        }
+        throw new Error('SERVER_URL or FILE_SERVER_URL must be set in environment variables, or provide serverUrl parameter')
+    }
+}
+
+export async function upload ({ serverUrl, cookie, files, meta, attach }: UploadOptions): Promise<UploadResponse> {
+    let FormDataClass: typeof FormData
+    let form: FormData | any
+
+    if (typeof window !== 'undefined') {
+        FormDataClass = FormData
+        form = new FormDataClass()
+    } else {
+        const FormDataModule = require('form-data')
+        FormDataClass = FormDataModule
+        form = new FormDataClass()
+    }
+
+    if (!Array.isArray(files) || files.length === 0) {
+        throw new Error('upload: "files" must be a non-empty array')
+    }
+
+    for (const f of files) {
+        if (typeof Buffer !== 'undefined') {
+            form.append('file', f, 'file.bin')
```

Comment:
Is it ok that all files have 'file.bin' name? Mb is it worth giving unique names?

Reply:
That's not really necessary here, because if user is not give us 'originalFileName' on the upload state - he don't might what actually it should be called. But it's okay to change it to the random one